### ISSUE TITLE
Get accessible name from element

### ISSUE THREAD

**Issue by dmurvihill** [testing-library/dom-testing-library#1378]
In keeping with [Guiding Principle #2](https://testing-library.com/docs/guiding-principles), I would like to assert what accessible name will be emitted for an element by screen readers and such. Currently this task, which all users of `testing-library` should be doing, requires implementing a [very complicated algorithm](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/) or relying on a third-party dependency like the archived [accname](https://www.npmjs.com/package/accname) package.

### Suggested implementation:

`getAccessibleName(myElement)`

### Workaround:
```typescript
expect(screen.getByRole('button', { name: expectedName })).toBe(myElement);
```

This gives the correct result, but in the failure case the debugging information is unsatisfactory ("expected x, received null" instead of showing the expected and actual name side-by-side).

**Comment by MatanBobi** [comment]
We're using the `dom-accessibility-api` package for that. Exporting it is just an implementation detail, the workaround you presented is actually what we want people to do. You can install [`dom-accessibility-api`](https://github.com/eps1lon/dom-accessibility-api) for it :)

**Comment by dmurvihill** [comment]
Thanks, that's very helpful. Do you think you could mention this in the docs? That package is hard to find by Web searching atm.
